zerver/webhooks/clickup/api_endpoints.py

```diff
@@ -0,0 +1,53 @@
+import requests
+from zerver.lib.outgoing_http import OutgoingSession
+from typing import Any, Dict
+
+
+class Error(Exception):
+    pass
+class APIUnavailableError(Error):
+    pass
+class BadRequestError(Error):
+    pass
+
+class ClickUpSession(OutgoingSession):
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(role="clickup", timeout=5, **kwargs)
+
+def make_clickup_request(path:str, api_key: str) -> dict:
+    headers: Dict[str, str] = {
```

Comment:
From a security standpoint, it would be good to have validation that `path` contains only expected characters for an ID -- not `/`s, for example, but an allowlist seems more correct than a blocklist.

Reply:
I added a function to help verify the URL path but coulnd't push it in this commit. I'll update it in the next commit